cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -259,6 +259,8 @@ public Future<Void> reconcile(KafkaStatus kafkaStatus, Clock clock)    {
                 .compose(i -> jmxSecret())
                 .compose(i -> podDisruptionBudget())
                 .compose(i -> podDisruptionBudgetV1Beta1())
+                .compose(i -> maybeDeleteStatefulSet())
+                .compose(i -> maybeDeletePodSet())
```

Comment:
It's a shame that the names don't reflect that the deletions are specific to migration. I can't think of a better suggestion though. `migrateFromStatefulSetController` and `migrateFromPodSetController` perhaps?

Reply:
Sounds like a good idea. I will rename them.